src/pools/balancer/Pool.ts

```diff
@@ -138,7 +138,7 @@ export class Pool {
 
     try {
       result = await token.methods
-        .approve(spender, new BigNumber(await this.amountToUnits(tokenAddress, amount)))
+        .approve(spender, Web3.utils.toWei(amountFormatted))
```

Comment:
i think amountToUnits needs some adjustments also if sending before convering to wei

Reply:
ah, yes, my bad, this is not done, there are other issues as well :)